marconi-chain-index/test/Spec/Marconi/ChainIndex/Indexers/Utxo/UtxoIndex.hs

```diff
@@ -3,142 +3,236 @@
+-- Assumption:  SQLite vacuum is disabled, so that we can accouhnt for generated Spents
+-- Note:        We expect this test to fail in this branch.
+allqueryUtxosShouldBeUnspent :: Property
+allqueryUtxosShouldBeUnspent = property $ do
+  events <- forAll genShelleyEraUtxoEvents -- we need to use shelley era addresses only to allow for point (4)
+  let numOfEvents = length events
+  -- We choose the `depth` such that we can prove the boundery condtions, see point (3).
+  -- this will ensure we have adequate coverage where events are in both, in-memory store and SQLite store
+  depth <- forAll $ Gen.int (Range.constantFrom (numOfEvents - 1) 1 (numOfEvents + 1))
+  Hedgehog.classify "Query both in-memory and storage " $ depth < numOfEvents
+  Hedgehog.classify "Query in-memory only" $ depth > numOfEvents
+
+  -- It is crtical that we perform NO vacuum.
+  -- With depth at such small numbers, the likelyhood of SQLite vaccume is almost certain in
+  indexer <- liftIO (Utxo.open ":memory:" (Utxo.Depth depth) False >>= Storable.insertMany events )
+  let
+    addressQueries :: [StorableQuery Utxo.UtxoHandle] -- we want to query for all addresses
+    addressQueries
+      = List.nub
+      . fmap (Utxo.UtxoAddress . Utxo._address)
+      . concatMap (Set.toList . Utxo.ueUtxos)
+      $ events
+  results <- liftIO . traverse (Storable.query Storable.QEverything indexer) $ addressQueries
+  let retrievedUtxoRows :: [Utxo.UtxoRow] = concatMap (\(Utxo.UtxoResult rs) -> rs ) results
+      txinsFromRetrievedUtsoRows :: [C.TxIn]  -- get all the TxIn from quried UtxoRows
+        = retrievedUtxoRows & each %~ (\r ->
+                                       C.TxIn (r ^. Utxo.urUtxo . Utxo.txId)(r ^. Utxo.urUtxo . Utxo.txIx))
+      txInsFromGeneratedEvents :: [C.TxIn]    -- get all the TxIn from quried UtxoRows
+        = concatMap (\(Utxo.UtxoEvent _ ins _ ) -> Set.toList ins ) events
+
+  -- A property of the generator is that there is at least one unspent transaction
+  -- this property also ensures that the next test will not succeed for the trivila case
+  -- when retrievedUtxoRows is an empty list
+  Hedgehog.assert (not . null $ retrievedUtxoRows)
+
+-- There should be no `Spent` in the retrieved UtxoRows
+  Hedgehog.footnote "Regression test must return at least one Utxo. Utxo's may not have any Spent in the Orig. event"
+  Hedgehog.assert
+    $ all (== True)
+    [u `notElem` txInsFromGeneratedEvents| u <- txinsFromRetrievedUtsoRows]
+
+-- | Round trip UtxoEvents to UtxoRow conversion
+-- The purpose of this test is to show that there is a isomorphism between `UtxoRow` and UtxoEvent.
+propRoundTripEventsToRowConversion :: Property
+propRoundTripEventsToRowConversion  = property $ do
+  events <- forAll UtxoGen.genUtxoEvents
+  let
+    txInsMap :: Map C.SlotNo (Set C.TxIn)
+    txInsMap
+      = Data.Map.fromList
+      . concatMap g
+      $ events
```

Comment:
Same comment as above. Don't use function composition if you already have the argument to be applied